electron/libelectron/src/backends/memory_mgmt.cc

```diff
@@ -0,0 +1,123 @@
+#include "memory_mgmt.hh"
+namespace Memory_mgmt{
+
+void Memory_mgmt::init_devmem(size_t dev_mem_size) {
+
+  /* Initialize structures*/
+  device_memory = malloc(dev_mem_size);
+  heap_start = (memBlock *)device_memory;
+  heap_start->size = dev_mem_size-sizeof(heap_start);
+  heap_start->used = 0;
+  heap_start->ptr = device_memory+sizeof(struct memBlock);
+
+  heap_start->prev = heap_start;
+  heap_start->next = heap_start;
```

Comment:
Is there a reason to keep this as a circular linked list?

Reply:
Thought it might be easier in case if we wanted to switch to next fit (Circular first fit approach)